wild-swimming-app/src/feature/map/Map.tsx

```diff
@@ -1,5 +1,105 @@
+import { MapContainer, TileLayer, Marker, Popup } from "react-leaflet";
+import React, { useState, useRef } from "react";
+import L from "leaflet";
+import "leaflet/dist/leaflet.css";
+
+import { getSewage } from "./utils/riversTrustAPI";
+import useGeoLocation from "./utils/getUserLocation";
+
+//import { fetchPostcodes } from "./utils/api";
 const Map = () => {
-  return <div></div>;
-};
+  const [center, setCenter] = useState({
+    lat: 51.5095146286,
+    lng: -0.1244828354,
+  });
+  const ZOOM_LEVEL = 9;
+  const mapRef = useRef();
+
+  const markerIcon = new L.Icon({
+    iconUrl:
+      "https://cdn4.iconfinder.com/data/icons/small-n-flat/24/map-marker-1024.png",
+    iconSize: [40, 40],
+    iconAnchor: [17, 46],
+    popupAnchor: [0, -46],
+  });
+
+  const userIcon = new L.Icon({
+    iconUrl: "https://img.icons8.com/?size=512&id=48096&format=png",
+    iconSize: [40, 40],
+    iconAnchor: [17, 46],
+    popupAnchor: [0, -46],
+  });
 
+  const location = useGeoLocation();
+  const showMyLocation = () => {
+    console.log(location);
+    if (location.loaded) {
+      console.log(location);
+      console.log(mapRef);
+      mapRef.current.flyTo(
+        [location.coordinates.lat, location.coordinates.lng],
+        ZOOM_LEVEL,
+        { animate: true }
+      );
+    }
+  };
+
+  const [filteredMarkers, setFilteredMarkers] = useState([]);
+  getSewage().then((data) => {
+    const mapData = data.data.features;
+    // console.log(mapData);
```

Comment:
This looks great, just spotted a couple of console.logs that may want to come out before we go live.